app/bundles/EmailBundle/Model/EmailActionModel.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Mautic\EmailBundle\Model;
+
+use Mautic\CategoryBundle\Entity\Category;
+
+readonly class EmailActionModel
+{
+    public function __construct(
+        private EmailModel $emailModel
+    ) {
+    }
+
+    public function setCategory(array $emailsIds, Category $newCategory): array
+    {
+        $emails = $this->emailModel->getByIds($emailsIds);
```

Comment:
```suggestion
        $emails = $this->emailModel->getRepository->findBy(['id' => $emailsIds]);
```
Then the whole `getByIds()` method can be deleted.